### ISSUE TITLE
Cannot read property '1' of null

### ISSUE THREAD

**Issue by burzum** [perrin4869/gulp-i18next-conv#64]
I'm getting this error on Windows 10 using NodeJS 6.9.1 and the latest version of this plugin. It doesn't work with v8.4.0 either. Also there is no verbose output for gulp... so I have no idea from where the error really comes from. All it says is this:

```sh
Unhandled rejection Error in plugin 'gulp-i18next-conv'
Message:
    Cannot read property '1' of null
```

Any idea how to fix it?

**Comment by perrin4869** [comment]
Could you check if i18next-conv (without gulp) works? I doubt the problem
is in this plugin proper since it delegates the hard work to i18next-conv

**Comment by burzum** [comment]
It works just fine:

```sh
λ i18next-conv -l en -s .\src\Locale\en\wa.po .\webroot\locales\en\wa.json
start converting
--> reading file from: .\src\Locale\en\wa.po
<-- writing file to: src\Locale\en\en\wa.json
```

My global install of it is v4.0.3.

**Comment by perrin4869** [comment]
I'll take a look into it later then

**Comment by perrin4869** [comment]
@burzum My guess is that the problem is in https://github.com/perrin4869/gulp-i18next-conv/blob/master/src/index.js#L15
First of all, I think I'll rename it to `determineLocale` and release as a breaking change, considering that `node-gettext` fixed their inconsistencies.
In your case, you probably want to pass in a custom `determineDomain` (or `determineLocale` after I release a new version) that accounts for your own setup

**Comment by perrin4869** [comment]
I'll try to log out a more helpful message in case `determineLocale` fails

**Comment by perrin4869** [comment]
@burzum Just pushed v3.0.0, try it out and tell me if it fixes your issues :D

**Comment by burzum** [comment]
Thank you @perrin4869. Unfortunatelly I'm still getting an error when running my gulp task:

```sh
Unhandled rejection Error in plugin 'gulp-i18next-conv'
Message:
    determineLocale failed
```

Is there something I can do to help you debugging the problem?

**Comment by perrin4869** [comment]
That is to be expected, at least now you get a more helpful message :)
Well, you need to provide a custom `determineLocale` option - that is a function that takes the po file path and contents, and extrapolates the locale from them. In your case, you probably wanna parse `.\src\Locale\en\wa.po` and get the right locale (`en`, I think)

**Comment by burzum** [comment]
OK, but what could be the reason that this works fine on Linux but not on Windows? If its really just about reading the files information.

**Comment by perrin4869** [comment]
Probably because paths are different. Try this:

```js
gulpI18next({
  determineLocale(filepath => {
    console.log(filepath);
    return 'en';
  });
});
```

Or something like that, and check if it's different in windows and linux

**Comment by burzum** [comment]
Stupid question, where do I put this in? In my gulpfile.js it's causing errors. I have zero experience developing on the node.js platform. I'm just using gulp.

**Comment by perrin4869** [comment]
Seems you deleted your previous comment :)
Figured it out? In case you haven't, here's a more complete version:

```js
// gulpfile.js
const i18next = require('gulp-i18next-conv');

gulp.task('i18next', () => {
  gulp.src('path/to/your/locales/*.po')
    .pipe(gulpI18next({
      determineLocale(filepath => {
        console.log(filepath);
        return 'en';
      });
    }))
    .pipe(gulp.dest('.'));
});
```

Give or take a parethesis here and there lol

**Comment by burzum** [comment]
It seems like its working. My laptop had an older node version. I need to test this on the machine that caused the problem originally. But sadly my mainboard died when the PSU blown up... I'm waiting to get new hardware. :( I hope to report you back until friday.

**Comment by burzum** [comment]
@perrin4869 OK, got my hardware fixed after they send me a new but broken PSU to replace my broken PSU... Trying to get this to work or debugged today.

Your code above gives me this error because I think the shell or node doesn't recognize ES6 for whatever reason.

```shell
λ gulp serve
C:\xampp\htdocs\wa3\gulpfile.js:35
      determineLocale(filepath => {
                               ^^

SyntaxError: Unexpected token =>
    at createScript (vm.js:74:10)
    at Object.runInThisContext (vm.js:116:10)
    at Module._compile (module.js:537:28)
    at Object.Module._extensions..js (module.js:584:10)
    at Module.load (module.js:507:32)
    at tryModuleLoad (module.js:470:12)
    at Function.Module._load (module.js:462:3)
    at Module.require (module.js:517:17)
    at require (internal/module.js:11:18)
    at Liftoff.handleArguments (C:\Users\Florian\AppData\Roaming\npm\node_modules\gulp\bin\gulp.js:116:3)
```

**Comment by perrin4869** [comment]
Nothing is being output by `determineLocale`?

**Comment by burzum** [comment]
I don't think it goes into that piece of code at all. https://gist.github.com/burzum/a6ab36a12925e2a20667e6f2211fd924#file-gulp-js-L32-L39

Here is the output when I run `gulp serve`:

```sh
C:\xampp\htdocs\wa3 [feature/201 +4 ~6 -0 !]
λ gulp
[14:41:27] Using gulpfile C:\xampp\htdocs\wa3\gulpfile.js
[14:41:27] Starting 'js-libs'...
[14:41:27] Starting 'css-libs'...
[14:41:27] Starting 'copyfonts'...
[14:41:27] Finished 'copyfonts' after 2.5 ms
[14:41:27] Starting 'copysprites'...
[14:41:27] Finished 'copysprites' after 649 μs
[14:41:27] Starting 'docs'...
[14:41:27] Finished 'docs' after 5.1 ms
[14:41:27] Starting 'js-lint'...
[14:41:27] Starting 'jquery:clean'...
[14:41:27] Starting 'jquery:generate'...
[14:41:27] Starting 'css-wa'...
[14:41:27] Starting 'delete-ckeditor'...
[14:41:27] Finished 'jquery:clean' after 55 ms
[14:41:29] Tested 3 tests, 3 passes, 0 failures: PASS
[14:41:35] Finished 'css-wa' after 8 s
[14:41:35] Starting 'minifiy-css-wa'...
[14:41:35] Tested 6 tests, 6 passes, 0 failures: PASS
[14:41:36] Tested 6 tests, 6 passes, 0 failures: PASS
[14:41:36] Tested 2 tests, 2 passes, 0 failures: PASS
[14:41:37] Finished 'css-libs' after 9.87 s
[14:41:37] Finished 'minifiy-css-wa' after 1.52 s
[14:41:37] Tested 9 tests, 9 passes, 0 failures: PASS
[14:41:38] Finished 'delete-ckeditor' after 10 s
[14:41:38] Starting 'setup-ckeditor'...
[14:41:40] Finished 'js-lint' after 13 s
[14:41:40] Tested 23 tests, 23 passes, 0 failures: PASS
[14:41:40] Tested 23 tests, 23 passes, 0 failures: PASS
[14:41:40] Finished 'js-libs' after 13 s
[14:41:40] Starting 'build-vendors'...
[14:41:40] Finished 'build-vendors' after 46 μs
[14:41:40] Finished 'jquery:generate' after 13 s
[14:41:40] Starting 'jquery:minify'...
[14:41:40] Starting 'jquery:i18n'...
Unhandled rejection Error in plugin 'gulp-i18next-conv'
Message:
    determineLocale failed

[14:41:40] Finished 'jquery:minify' after 248 ms
[14:41:46] Finished 'setup-ckeditor' after 8.48 s
[14:41:46] Starting 'cleanup-ckeditor'...
[14:41:46] Finished 'cleanup-ckeditor' after 60 ms
[14:41:46] Starting 'copy-ckeditor-config'...
[14:41:46] Starting 'build-ck-plugins'...
[14:41:46] Finished 'build-ck-plugins' after 587 μs
[14:41:46] Finished 'copy-ckeditor-config' after 16 ms
[14:41:46] Starting 'build-ckeditor'...
[14:41:46] Finished 'build-ckeditor' after 47 μs
C:\xampp\htdocs\wa3 [feature/201 +4 ~6 -0 !]
```

**Comment by perrin4869** [comment]
hm... that's weird, it seems like it `gulp-i18next-conv` is still using the default `determineLocale`, since there is no way the one I sent you would fail...
Still, there are tests checking that you can use a custom `determineLocale` which are passing.
Is there a chance you could try debugging this issue by editing the files inside `node_modules/gulp-i18next-conv` directly?

**Comment by perrin4869** [comment]
Can you check if `3.0.1` fixes this issue for you?

**Comment by perrin4869** [comment]
@burzum I think this issue has been resolved with `4.0.1`, if it is not then please reopen :)

**Comment by burzum** [comment]
@perrin4869 thanks a lot for fixing this. I can confirm that 4.0.1 works! 🎉
